src/libstore/globals.hh

```diff
@@ -973,6 +973,19 @@ public:
           | `~/.nix-profile`  | `$XDG_STATE_HOME/nix/profile`  |
           | `~/.nix-defexpr`  | `$XDG_STATE_HOME/nix/defexpr`  |
           | `~/.nix-channels` | `$XDG_STATE_HOME/nix/channels` |
+
+          If you already have Nix installed and are using Profiles or Channels, you should migrate manually when you enable this option.
+          To do so, create `$XDG_STATE_HOME/nix`, then move `.nix-profile`, `.nix-defexpr` and `.nix-channels` as specified in the table above.
+          If `$XDG_STATE_HOME` is not set, use `$HOME/.local/state/nix` instead of `$XDG_STATE_HOME/nix`.
+          This can be achieved with the following shell commands:
```

Comment:
We should be more specific about the *when*.

```suggestion
          This can be achieved by running the following shell commands before invoking `nix-channel`, `nix-env`, or `nix-profile` with the changed setting:
```

Reply:
Also link to the pages for the respective commands.

Reply:
It's more than those commands, and not all invocations of those need the migration. I think it's fine to not specify _when_, "big picture" is fairly obvious (you should do it after you change the setting), small details get complicated quickly.

Reply:
This is simply to orient users. We don't know when in their journey people will encounter this piece of writing, and what they will know about Nix at that point.